src/core/cwa.ts

```diff
@@ -76,6 +82,41 @@ export default class Cwa {
     })
   }
 
+  async initMercure(headers) {
+    if (this.eventSource) return
+    const link = headers.link
+    if (!link) {
+      console.warn('No Link header found.')
+      return
+    }
+
+    const match = link.match(/<(.*)>.*rel="mercure".*/);
+    if (!match || !match[1]) {
+      console.log('No mercure rel in link header.')
+      return
+    }
+
+    const hub = new URL(match[1])
+    hub.searchParams.append('topic', '/_/routes/{id}')
+    hub.searchParams.append('topic', '/_/pages/{id}')
+    hub.searchParams.append('topic', '/_/layout/{id}')
+    hub.searchParams.append('topic', '/_/component_collections/{id}')
+    hub.searchParams.append('topic', '/component/html_contents/{id}')
```

Comment:
We will need to ensure that we listen to whatever components are returned. This is dynamic depending on the application as in the API the user creates these objects by extending an abstract class. So we re receive components from the API we will need to subscribe to their updates is we aren't already.

Reply:
Indeed and we should keep a single Eventsource open and maybe close and reopen it if we need more topics. Something that could work is keeping the topics list in the state and when it changes reset the Eventsource.

Reply:
Latest merge will close and re-open on route change.